### ISSUE TITLE
[BUG] Multiline CloudBlockBlob.Properties.ContentDisposition causes HTTP 403 error on blob write

### ISSUE THREAD

**Issue by pvones** [Azure/azure-sdk-for-net#9009]
WindowsAzure.Storage package version 9.3.3
.NET Framework 4.8

Typically the *System.Net.Mime.ContentDisposition* class is used to build the content disposition string to assign to the *CloudBlockBlob.Properties.ContentDisposition* property.

If the *ContentDisposition.FileName* property contains **non-english** characters, the content disposition string is UTF-8 and Base64 encoded. When the filename **is longer than 41 characters**, the *ContentDisposition.ToString()* method wraps it by a new text line (CR/LF characters). It fails with HTTP 403 error in this case when the write stream is closed/flushed.

```csharp
CloudStorageAccount account = CloudStorageAccount.Parse("ConnectionString");
CloudBlobClient client = account.CreateCloudBlobClient();
CloudBlobContainer container = client.GetContainerReference("test001");
container.CreateIfNotExists();

CloudBlockBlob blob = container.GetBlockBlobReference("Blob name");
ContentDisposition cd = new ContentDisposition();
cd.FileName = "File ěščřžýáíé long long long long long long.txt";
blob.Properties.ContentDisposition = cd.ToString();

using (Stream stream = blob.OpenWrite())
{
  byte[] data = Encoding.ASCII.GetBytes("Text");
  stream.Write(data, 0, data.Length);
}
```

It throws *StorageException* with message "The remote server returned an error: (403) Forbidden."

Because the ContentDisposition string is delimited by CR/LF for UTF-8 encoded filenames longer than 41 characters:
```
"attachment; filename=\"=?utf-8?B?RmlsZSDEm8WhxI3FmcW+w73DocOtw6kgbG9uZyBsb25nIGxvbmcgbG9u?=\r\n =?utf-8?B?ZyBsb25nIGxvbmcudHh0?=\""
```

**Comment by amnguye** [comment]
Hi,

What version of the SDK are you using? I see that you're using CloudBlockBlob which is a object used in version 11 of the SDK but is no longer used in 12 (the current version).

I would try the current version of the SDK to see if this problem gets resolved.

**Comment by kartikshah9** [comment]
@pvones can you provide inputs to the above query?

**Comment by pvones** [comment]
The latest version of the package I can update to is **9.3.3**. Versions 9.4.0 and above have **completely different interface** that would require to refactor all code with no added value provided. The backward compatibility is still imporant to consider when you actualy use the code (not just playing with it).

**Comment by amnguye** [comment]
I apologize for the late extremely late reply.

I made a new issue for you on the old repo(https://github.com/Azure/azure-storage-net/issues/968), which is the version you are currently using. A 403 is an odd error considering that its forbidden, instead of a bad request if we formatted the name bad.

Looks like we resolved this issue in later versions, however I understand your concern about upgrading.
